cpp/ql/src/Likely Bugs/Memory Management/UninitializedLocal.ql

```diff
@@ -86,5 +86,10 @@ from
 where
   conf.hasFlowPath(source, sink) and
   isSinkImpl(sink.getInstruction(), va) and
-  v = va.getTarget()
+  v = va.getTarget() and
+  (
+    exists(Call c | c.getQualifier() = va)
+    implies
+    exists(Call c | c.getQualifier() = va and not c.getTarget().isStatic())
+  )
```

Comment:
For performance reasons this should be moved into the `isSinkImpl` predicate. Then it will be used to reduce the set of sinks considered by the "MustFlow" dataflow library and thus make the whole query a tiny bit faster.

Reply:
In fact there appears to be a `commonException()` predicate for this purpose.

---

FWIW,  believe your logic is equivalent to just:
```
forall(Call c | c.getQualifier() = va | c.getTarget().isStatic())
```
I'm not sure when there would be more than one call with the same qualifying expression.  It seems like we could just say:
```
not exists(Call c | c.getQualifier() = va | c.getTarget().isStatic())
```

Reply:
> In fact there appears to be a commonException() predicate for this purpose.

Indeed, I think sticking this new logic inside `commonException()` is the right way to go.

I also completely agree that we should replace the implication-style logic with `not exists(Call c | c.getQualifier() = va | c.getTarget().isStatic())`.

Or rather, if we stick this code into `commonException` it should be `exists(Call c | c.getQualifier() = va | c.getTarget().isStatic())`

Reply:
Fixed